rust/sdk-processor/src/steps/common/parquet_buffer_step.rs

```diff
@@ -0,0 +1,387 @@
+use crate::{
+    parquet_processors::{ParquetTypeEnum, ParquetTypeStructs},
+    steps::common::gcs_uploader::Uploadable,
+};
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    traits::{
+        pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
+    },
+    types::transaction_context::{TransactionContext, TransactionMetadata},
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use mockall::mock;
+use std::{collections::HashMap, time::Duration};
+use tracing::debug;
+
+/// `ParquetBuffer` is a struct that holds `ParquetTypeStructs` data
+/// and tracks the buffer size in bytes, along with metadata about the data in the buffer.
+struct ParquetBuffer {
+    pub buffer: ParquetTypeStructs,
+    pub buffer_size_bytes: usize,
+    current_batch_metadata: Option<TransactionMetadata>,
+}
+
+impl ParquetBuffer {
+    fn new(parquet_type: &ParquetTypeEnum) -> Self {
+        Self {
+            buffer: ParquetTypeStructs::default_for_type(parquet_type),
+            buffer_size_bytes: 0,
+            current_batch_metadata: None,
+        }
+    }
+
+    /// Updates the metadata of the internal buffer with the latest information from `cur_batch_metadata`.
+    /// This is used to track the end version and timestamp of the batch data in the buffer.
+    pub fn update_current_batch_metadata(&mut self, cur_batch_metadata: &TransactionMetadata) {
+        if let Some(buffer_metadata) = &mut self.current_batch_metadata {
+            // Update metadata fields with the current batch's end information
+            buffer_metadata.end_version = cur_batch_metadata.end_version;
```

Comment:
Before we set, the new versions, should we add some validation that there are not gaps in data (buffer_metadata.end_version + 1 = cur_batch_metadata.start_version? It seems like everything is parsed in order 🤔

Reply:
yeah it shouldn't happen but it's good to add a safety here!

Reply:
Let's try to avoid panics if possible and return a Result<ProcessorError>